src/Broker/Broker.c

```diff
@@ -107,7 +108,11 @@ Broker_initialize(enum SystemBus valid_buses[SYSTEM_BUSES_NUMBER])
             // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=85398
             enum PacketizerCfg packetizer_type = bus_to_packetizer_cfg[bus_id];
             packetizer_init_function packetizer_init = packetizers_functions[packetizer_type].init;
-            packetizer_init(&packetizers_data[bus_id], bus_id);
+            packetizer_init(&packetizers_data[bus_id], bus_id, &headerSize);
+
+            if(packetizer_type == PACKETIZER_DEVICE_PROVIDED) {
+                packetizers_functions[packetizer_type].headerSize = headerSize;
```

Comment:
This is a simpler solution than the one we discussed - does this one handle a situation in which there are two devices, with different packetizers? Won't the last one overwrite the size reported by the previous ones?

Reply:
Added extra array with CD packetizer header, the header size is get by busId in DeviceProvidedPacketizer